source/core/NumericString.cpp

```diff
@@ -2165,18 +2178,22 @@ VIREO_FUNCTION_SIGNATURET(StringToByteArray, StringToByteArrayParamBlock)
         return _NextInstruction();
     }
     const Int32 stringLength = stringIn->Length();
+    ConstCStr functionName = "String To Byte Array";
     if (errorCluster != nullptr) {
         // If the user did not pass errorCluster, no need to validate the string, as we are copying
         // to byteArrayOut unconditionally
-        CheckValidUTF8Error(stringIn->RawBegin(), stringIn->RawBegin() + stringLength, errorCluster);
+        CheckValidUTF8Error(stringIn->RawBegin(), stringIn->RawBegin() + stringLength, errorCluster, functionName);
     }
     if (byteArrayOut != nullptr) {
         IntIndex rank = byteArrayOut->Rank();
         VIREO_ASSERT(rank == 1);
         TypeRef eltType = byteArrayOut->ElementType();
         if (!eltType->IsA(&TypeCommon::TypeUInt8)) {
```

Comment:
Based on the other comment this may also update to a harderror instead of a runtime error

Reply:
Yes, NXG must guarantee this is TypeUInt8

Reply:
Done in commit 3.